### DISCUSSION TITLE
Part 3 exercise statement clarification

### DISCUSSION THREAD

**Discussion by bruluk5w** [sebastian-hofstaetter/teaching#75] [Advanced IR: Exercise Q&A] [answered]
Hi,
The 3rd bullet point in the exercise statement of  part 3 states "...store results with the HuggingFace library". In which format should the results be stored? What is "the HuggingFace library"? The only library that I think could be meant is `datasets` (https://huggingface.co/docs/datasets). Or does `transformers` have specific capabilities for IO that I missed?
Edit:
It is also specifically asked to "tokenize query passage pairs". As far as I saw, the transformers qa pipeline does this already automatically. Is there a specific way that we are expected to follow?

**Comment by DezignDezire** [comment]
The qa pipeline is only for inference (predicting answers), but you can not use it for training.

**Reply by bruluk5w** [reply]
Yes, but then the exercise statement needs clarification because it only states that we should run inference. There is no mention of fine-tuning the model.

**Reply by sophiaalthammer** [reply]
Yes you shouldnt do any training only inference.

**Comment by bruluk5w** [comment]
Anyone? From the lecture team maybe? @sebastian-hofstaetter

**Comment by sophiaalthammer** [comment, marked as answer]
Sorry for the late response! 
The exercise statement could be clarified like that:
Implement code in src/extractive_qa.py to load the model, tokenize query passage pairs, and run inference, store results. Use the HuggingFace library for the implementation.

You can decide the format in which you store it yourself. It is supposed that you use the automatic tokenization from the HuggingFace libary.

**Reply by bruluk5w** [reply]
Thank you :)
